subprojects/core/src/main/java/org/gradle/normalization/internal/DefaultRuntimeClasspathNormalization.java

```diff
@@ -21,24 +21,71 @@
 import org.gradle.api.internal.changedetection.state.IgnoringResourceFilter;
 import org.gradle.api.internal.changedetection.state.ResourceFilter;
 
+import java.util.Locale;
+import java.util.concurrent.atomic.AtomicBoolean;
+
 public class DefaultRuntimeClasspathNormalization implements RuntimeClasspathNormalizationInternal {
     private final ImmutableSet.Builder<String> ignoresBuilder = ImmutableSet.builder();
+    private final ImmutableSet.Builder<String> attributeIgnoresBuilder = ImmutableSet.builder();
+    private final ImmutableSet.Builder<String> propertiesIgnoresBuilder = ImmutableSet.builder();
+    private AtomicBoolean evaluated = new AtomicBoolean();
     private ResourceFilter resourceFilter;
+    private ResourceFilter attributeResourceFilter;
+    private ResourceFilter propertyResourceFilter;
 
     @Override
     public void ignore(String pattern) {
-        if (resourceFilter != null) {
-            throw new GradleException("Cannot configure runtime classpath normalization after execution started.");
-        }
+        checkNotEvaluated();
         ignoresBuilder.add(pattern);
     }
 
     @Override
-    public ResourceFilter getResourceFilter() {
-        if (resourceFilter == null) {
+    public void ignoreManifestAttribute(String pattern) {
+        checkNotEvaluated();
+        attributeIgnoresBuilder.add(pattern.toLowerCase(Locale.ROOT));
+    }
+
+    @Override
+    public void ignoreManifestProperty(String pattern) {
+        checkNotEvaluated();
+        propertiesIgnoresBuilder.add(pattern);
+    }
+
+    @Override
+    public RuntimeClasspathResourceFilters getResourceFilters() {
```

Comment:
Grouping them is for the downstream use of the class - it avoids needing three (and maybe later more) constructor arguments for the filters.

The `evaluated` state seems necessary to me, and it's atomic and thread safe? We don't want people adding normalization configuration at runtime and not realizing it's not taking affect?

Reply:
Configuration happens single threaded, so no need for synchronization here. As I mentioned in other places I don't think we need the grouping class.

Reply:
It allows one constructor argument instead of three for values that should be carried together - there's at least three classes along that path that already have very long method/constructor argument lists.